src/main/java/com/extremelyd1/listener/ItemListener.java

```diff
@@ -53,4 +56,55 @@ public void onPlayerDropItem(PlayerDropItemEvent e) {
             e.setCancelled(true);
         }
     }
+
+    /**
+     * Caters for collecting milk, water or lava
+     *
+     * @param e - PlayerBucketFillEvent
+     */
+    @EventHandler
+    public void onBucketFill(PlayerBucketFillEvent e) {
+        if (!game.getState().equals(Game.State.IN_GAME)) {
+            e.setCancelled(true);
+            return;
+        }
+
+        game.onMaterialCollected(e.getPlayer(), e.getItemStack().getType());
+    }
+
+    /**
+     * This event caters for the collection of fishes into buckets, and the collection of mushroom stew from mooshrooms
+     *
+     * @param e - PlayerInteractAtEntityEvent
+     */
+    @EventHandler
+    public void onPlayerInteractAtEntity(PlayerInteractAtEntityEvent e) {
+        if (e.isCancelled()) return;
+
+        Map<EntityType, Material> fishes = new HashMap<EntityType, Material>() {{
```

Comment:
These hashmaps could be better statically defined at the top of the class rather than in this method body.

Reply:
I have statically defined them at the top of this class